cron-job-utils/src/com/axonivy/utils/cronjobutils/CronByGlobalVariableTriggerStartEventBean.java

```diff
@@ -0,0 +1,195 @@
+	public void start(IProgressMonitor monitor) throws ServiceException {
+		super.start(monitor);
+		if (scheduler != null && trigger != null) {
+			try {
+				scheduler.start();
+			} catch (SchedulerException e) {
+				throw new ServiceException(e);
+			}
+		}
+	}
+
+	@Override
+	public void stop(IProgressMonitor monitor) throws ServiceException {
+		super.stop(monitor);
+		if (scheduler != null) {
+			try {
+				scheduler.shutdown();
+			} catch (SchedulerException e) {
+				throw new ServiceException(e);
+			}
+		}
+	}
+
+	@Override
+	public void execute(final JobExecutionContext context) throws JobExecutionException {
+		if (context.getJobDetail().getJobDataMap().containsKey(RUNTIME_KEY)) {
+			final IProcessStartEventBeanRuntime eventRuntime = (IProcessStartEventBeanRuntime) context.getJobDetail()
+					.getJobDataMap().get(RUNTIME_KEY);
+			
+			if (eventRuntime != null) {
+				final Logger log = eventRuntime.getRuntimeLogLogger();
+				final String triggerIdentifier = context.getTrigger().getJobKey().getName();
+				Throwable throwable = null;
+
+				Long jobStartTs = startedJobs.get(triggerIdentifier);
+				String nextRun = String.format("Next fire at %1$tF %1$tT.", context.getTrigger().getNextFireTime());
+				if (jobStartTs != null) {
+					log.warn(
+							"Not starting job {0}, since an instance is currently running (the instance is running since {1} ms). {2}",
+							triggerIdentifier, System.currentTimeMillis() - jobStartTs, nextRun);
+				} else {
+					log.info("Starting job {0}", triggerIdentifier);
+					long startTs = System.currentTimeMillis();
+					startedJobs.put(triggerIdentifier, startTs);
+					try {
+						eventRuntime.executeAsSystem(new Callable<Void>() {
+							@Override
+							public Void call() throws Exception {
+								String firingReason = "Cron Trigger started " + triggerIdentifier;
+								Map<String, Object> parameters = new HashMap<>();
+								eventRuntime.fireProcessStartEventRequest(null, firingReason, parameters);
+								return null;
+							}
+						});
+					} catch (Throwable t) {
+						log.error(
+								"Exception while trying to execute cron job {0}. Note, the exception might have been shown already.",
+								t, triggerIdentifier);
+						throwable = t;
+					} finally {
+						startedJobs.remove(triggerIdentifier);
+					}
+					
+					long endTs = System.currentTimeMillis();
+					String stats = String.format("execution time %.3f", (endTs - startTs) / 1000.0);
+					if (throwable != null) {
+						log.error("Job {0} ended with error {1} ({2})", triggerIdentifier, throwable, stats);
+					} else {
+						log.info("Job {0} ended normally ({1})", triggerIdentifier, stats);
+					}
+				}
+			}
+		}
+	}
+
+	
+	/**
+	 * Editor class to work with the configuration.
+	 */
+	public static class Editor extends UiEditorExtension {
+
+		private IUiFieldEditor globalVariable;
+
+		
+		@Override
+		public void initUiFields(ExtensionUiBuilder ui) {
+			ui.label("Global Variable name").create();

```

Comment:
We no longer call that concept 'global variables' : just 'Variables' -> https://dev.axonivy.com/doc/10.0/engine-guide/reference/engine-cockpit/configuration.html#variables